Source/qol/itemlabels.cpp

```diff
@@ -0,0 +1,162 @@
+/**
+* @file itemlabels.cpp
+*
+* Adds item labels QoL feature
+*/
+
+#include <vector>
+#include <string>
+#include <unordered_set>
+
+#include "inv.h"
+#include "gmenu.h"
+#include "cursor.h"
+#include "common.h"
+#include "control.h"
+#include "itemlabels.h"
+#include "engine/render/cel_render.hpp"
+
+namespace devilution {
+
+namespace {
+
+struct itemLabel {
+	int id, width;
+	Point pos;
+	std::string text;
+};
+
+std::vector<itemLabel> labelQueue;
+
+bool altPressed = false;
+bool isLabelHighlighted = false;
+std::array<std::optional<int>, ITEMTYPES> labelCenterOffsets;
+bool invertHighlightToggle = false;
+
+} // namespace
+
+void ToggleItemLabelHighlight()
+{
+	invertHighlightToggle = !invertHighlightToggle;
+}
+
+void AltPressed(bool pressed)
+{
+	altPressed = pressed;
+}
+
+bool IsItemLabelHighlighted()
+{
+	return isLabelHighlighted;
+}
+
+void AddItemToLabelQueue(int id, int x, int y)
+{
+	if (altPressed == invertHighlightToggle)
+		return;
+	ItemStruct *it = &items[id];
+
+	char textOnGround[64];
+	if (it->_itype == ITYPE_GOLD)
+		std::sprintf(textOnGround, "%i gold", it->_ivalue);
+	else
+		std::strcpy(textOnGround, it->_iIdentified ? it->_iIName : it->_iName);
+
+	int nameWidth = GetLineWidth(textOnGround);
+	BYTE index = ItemCAnimTbl[it->_iCurs];
+	if (!labelCenterOffsets[index].has_value()) {
+		std::pair<int, int> itemBounds = MeasureSolidHorizontalBounds(*it->_iAnimData, it->_iAnimFrame);
+		labelCenterOffsets[index].emplace((itemBounds.first + itemBounds.second) / 2);
+	}
+
+	x += *labelCenterOffsets[index];
+	y -= TILE_HEIGHT;
+	if (!zoomflag) {
+		x <<= 1;
+		y <<= 1;
+	}
+	x -= nameWidth >> 1;
+	labelQueue.push_back(itemLabel { id, nameWidth, { x, y }, textOnGround });
+}
+
+bool IsMouseOverGameArea()
+{
+	if ((invflag || sbookflag) && MouseX > RIGHT_PANEL && MouseY <= SPANEL_HEIGHT)
+		return false;
+	if ((chrflag || questlog) && MouseX < SPANEL_WIDTH && MouseY <= SPANEL_HEIGHT)
+		return false;
+	if (MouseY >= PANEL_TOP && MouseX >= PANEL_LEFT && MouseX <= PANEL_LEFT + PANEL_WIDTH)
+		return false;
+
+	return true;
+}
+
+void FastDrawHorizLine(const CelOutputBuffer &out, int x, int y, int width, Uint8 col)

```

Comment:
This is identical to `UnsafeDrawHorizontalLine`, but probably better to use `DrawHorizontalLine` as you do not appear to be doing any bound checks.

Reply:
@qndel You marked this resolved but the code is still the same. Forgot to push?

Reply:
no, was poking :P